sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -107,6 +110,91 @@ public CompletionStage<Response> executeQueryAsync(Query query) {
     return observer.asFuture();
   }
 
+  @Override
+  public CompletionStage<Response> executeQueryAsync(
+      String keyspaceName,
+      String tableName,
+      Function<Optional<Schema.CqlTable>, Query> queryProducer) {
+    String decoratedKeyspaceName = addTenantPrefix(keyspaceName);
+    CqlKeyspaceDescribe keyspace = keyspaceCache.getIfPresent(decoratedKeyspaceName);
+    if (keyspace == null) {
+      // Nothing in our local cache, check if the bridge has a more recent version
+      return forceFetchAndExecute(keyspaceName, decoratedKeyspaceName, tableName, queryProducer);
+    } else {
+      // Build the query with our local version
+      Query query;
+      try {
+        query = produceQuery(queryProducer, Optional.of(keyspace), tableName);
+      } catch (Throwable t) {
+        // Always retry with the latest version, in case the error is caused by stale metadata
+        // (e.g. using a table that has not yet replicated to our local cache)
+        return forceFetchAndExecute(keyspaceName, decoratedKeyspaceName, tableName, queryProducer);
```

Comment:
Ah. This is where retry is for case of query building failing, possibly due to stale schema. And calls different method